gapic-generator/test/gapic/formatting_utils_test.rb

```diff
@@ -239,26 +239,112 @@ def test_escape_braces_separators_capitalized
-    result = Gapic::FormattingUtils.escape_braces ["hello {ObjectName#method OPTIONAL_TITLE}\n"]
+    result = Gapic::FormattingUtils.format_doc_lines nil, ["hello {ObjectName#method OPTIONAL_TITLE}\n"]
     assert_equal ["hello \\\\{ObjectName#method OPTIONAL_TITLE}\n"], result
   end
 
   def test_escape_braces_yardexample_class
-    result = Gapic::FormattingUtils.escape_braces ["hello {Class::CONSTANT My constant's title}\n"]
+    result = Gapic::FormattingUtils.format_doc_lines nil, ["hello {Class::CONSTANT My constant's title}\n"]
     assert_equal ["hello \\\\{Class::CONSTANT My constant's title}\n"], result
   end
   
   def test_escape_braces_yardexample_method
-    result = Gapic::FormattingUtils.escape_braces ["hello {#method_inside_current_namespace}\n"]
+    result = Gapic::FormattingUtils.format_doc_lines nil, ["hello {#method_inside_current_namespace}\n"]
     assert_equal ["hello \\\\{#method_inside_current_namespace}\n"], result
   end
 
+  def test_xref_message
+    api = FakeApi.new do |api|
+      api.add_file! "google.cloud.example" do
+        api.add_message! "Earth"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api, ["Hello, [World][google.cloud.example.Earth]!\n"]
+    assert_equal ["Hello, {Google::Cloud::Example::Earth World}!\n"], result
+  end
+
+  def test_xref_message_with_ruby_package
+    api = FakeApi.new do |api|
+      api.add_file! "google.cloud.example", "Google::Cloud::MyExample" do
+        api.add_message! "Earth"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api, ["Hello, [World][google.cloud.example.Earth]!\n"]
+    assert_equal ["Hello, {Google::Cloud::MyExample::Earth World}!\n"], result
+  end
+
+  def test_xref_multiple_messages
+    api = FakeApi.new do |api|
+      api.add_file! "google.cloud.example" do
+        api.add_message! "Earth"
+        api.add_message! "Ruby"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api,
+      ["Hello, [Ruby][google.cloud.example.Ruby] [World][google.cloud.example.Earth]!\n"]
+    assert_equal ["Hello, {Google::Cloud::Example::Ruby Ruby} {Google::Cloud::Example::Earth World}!\n"], result
+  end
+
+  def test_xref_proto_not_found
+    api = FakeApi.new do |api|
+      api.add_file! "google.cloud.example" do
+        api.add_message! "Earth"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api, ["Hello, [Ruby][google.cloud.example.Ruby]!\n"]
+    assert_equal ["Hello, [Ruby][google.cloud.example.Ruby]!\n"], result
+  end
+
+  def test_xref_service
+    api = FakeApi.new do |api|
+      api.add_file! "google.cloud.example" do
+        api.add_service! "Earth"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api, ["Hello, [World][google.cloud.example.Earth]!\n"]
+    assert_equal ["Hello, {Google::Cloud::Example::Earth::Client World}!\n"], result
+  end
+
+  def test_xref_with_method
+    api = FakeApi.new do |api|
```

Comment:
Missing a test for the Field out of the Service/Message/Method/Field switch block in transform_xref

Reply:
good catch. done.